sklearn/base.py

```diff
@@ -475,7 +491,9 @@ class ClassifierMixin:
     """
 
     # TODO(1.8): Remove this attribute
-    _estimator_type = "classifier"
+    @property
+    def _estimator_type(self):
+        return "classifier"
```

Comment:
Can't you just decorate the properties with @deprecated instead of defining `__new__` for the base class ?